### DISCUSSION TITLE
How do I set up code-server for debug ?

### DISCUSSION THREAD

**Discussion by ghost** [coder/code-server#2288] [Q&A] [answered]
This is frustrating me. I can not get the debug to work.
I'm using a Debian 10 machine.
I've installed code-server 3.6.2.
I've installed the neo-vim editor extension. It works.
I've installed rustup. I can run cargo, write and run Rust code using code-server within Safari.
For debug, I install the codelldb extension, version 1.6.0. No joy.
I can set a red dot break point but when I F5 to start the debug... 
"Activating extensions" arrow-spins on bottom status line.
and then nothing happens.
Well, actually, code-server remains as if I'd not tried to star the debugger.
Yes, I have made a launch.json. It, might be wrong.
so I ask,
What is a successful recipe for debug in code-server?

**Comment by code-asher** [comment, marked as answer]
Right now the extension host doesn't work at all on Safari. See
https://github.com/cdr/code-server/issues/2232#issuecomment-720608365

Currently the only workaround is to use 3.6.0 (or a different browser).
I'm hoping that VS Code 1.51.0 will resolve the issue and am currently
working on the upgrade.

**Reply by ghost** [reply]
On my laptop, MacBook pro running Catalina, I have Chrome installed.
Using Chrome there is a different failure(?):
"The debugger exited without completing startup handshake."

[errors.txt](https://github.com/cdr/code-server/files/5525939/errors.txt)
[more-errors.txt](https://github.com/cdr/code-server/files/5525963/more-errors.txt)

Let me regress my code-server image to 3.6.0 and see if that is a viable workaround.

**Reply by code-asher** [reply]
Curious, it's definitely possible you're actually running into a different issue. Going off those errors I'd say our marketplace isn't building vscode-lldb quite right, or isn't building it for other platforms like Mac.

**Reply by ghost** [reply]
Yup, I agree, there could be multiple issues. Hey, it's software.
I fumbled around yesterday trying to regress to code-server 3.6.0. 
I was not successful. 
Currently I have one Docker container running code-server 3.6.2 that works.
I'm stepping back from installing the debugger and will wait for code-server wrapping in VSCode 1.51.

Stay safe.
